### PR TITLE
Update recoveryservices SDK to 2022-10-01 version

### PR DESCRIPTION
**Pull request by zubairabid**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

We're planning to move our recoveryservices (and recoveryservicesbackup) SDK to stable releases. This SDK has been generated from a stable swagger version, as specified in [the release process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process). Is there anything else we are required to do?

### CODE REVIEW

File: sdk/recoveryservices/Microsoft.Azure.Management.RecoveryServices/src/Generated/Models/VaultProperties.cs [hunk 1]
```diff
@@ -50,11 +50,16 @@ public VaultProperties()
...
+        /// <param name="securitySettings">Security Settings of the
+        /// vault</param>
+        public VaultProperties(string provisioningState = default(string), UpgradeDetails upgradeDetails = default(UpgradeDetails), IList<PrivateEndpointConnectionVaultProperties> privateEndpointConnections = default(IList<PrivateEndpointConnectionVaultProperties>), string privateEndpointStateForBackup = default(string), string privateEndpointStateForSiteRecovery = default(string), VaultPropertiesEncryption encryption = default(VaultPropertiesEncryption), VaultPropertiesMoveDetails moveDetails = default(VaultPropertiesMoveDetails), string moveState = default(string), string backupStorageVersion = default(string), string publicNetworkAccess = default(string), MonitoringSettings monitoringSettings = default(MonitoringSettings), VaultPropertiesRedundancySettings redundancySettings = default(VaultPropertiesRedundancySettings), SecuritySettings securitySettings = default(SecuritySettings))
```
Comment (archerzz) [inline sdk/recoveryservices/Microsoft.Azure.Management.RecoveryServices/src/Generated/Models/VaultProperties.cs:62]: Breaking change. `publicNetworkAccess` is inserted in existing parameter list, which could cause breaking change for existing customer codes.

**Review by archerzz — CHANGES_REQUESTED** [review]
There are breaking changes in the interfaces. Here are 2 options:
  - Add manual codes to provide backward compatibility, see compute for reference: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations
- Or you need to bump the major version

**Comment by hiaga** [comment]
> @zubairabid We're migrating to Track2 SDKs. For existing Track1 SDK, by default we only allow to publish to dev feeds (e.g. not on `nuget.org`). Are you aware of that?

@archerzz : we have existing clients like PowerShell and CLI which are not yet migrated to track 2 SDKs. will we still not be allowed to add a new SDKs on top of existing ones like here ? - https://www.nuget.org/packages/Microsoft.Azure.Management.RecoveryServices.Backup/5.0.0-preview

**Comment by ArthurMa1978** [comment]
> > @zubairabid
> >             
> >             Zubair Abid
> >             FTE We're migrating to Track2 SDKs. For existing Track1 SDK, by default we only allow to publish to dev feeds (e.g. not on `nuget.org`). Are you aware of that?
> 
> @archerzz Mingzhe Huang FTE : we have existing clients like PowerShell which are not yet migrated to track 2 SDKs. will we still not be allowed to add a new SDKs on top of existing ones like here ? - https://www.nuget.org/packages/Microsoft.Azure.Management.RecoveryServices.Backup/5.0.0-preview

For PowerShell, please just update the sdk code but don't release a new package.
Contact PowerShell team for updating the cmdlet after the code merged.

**Review by ArthurMa1978 — APPROVED** [review]
This is only for PowerShell integration, don't use this PR to release a new Track 1 package

**Comment by hiaga** [comment]
@ArthurMa1978 : We need a stable SDK release with this PR, Could you please check

**Comment by Yao725** [comment]
@hiaga, please follow this [guide](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=sdk-generation-from-updated-spec) to update the track 2 SDK and then we can release it.

**Pull request merged**